triggers/daml/MatchingEngine.daml

```diff
@@ -91,3 +140,91 @@ handleOrder party orders (orderCid, order) = do
           timeMatched = show time
 
       return ()
+
+handleClearedOrders : Party -> TriggerA CurrentOrderId ()
```

Comment:
There is a pretty high amount of copied code here from the `Order` matching (since there's no typeclassing of templates...), I figure it can be cleaned up a bit after the main refactor is merged in.